File: benchmark/nodejs/polynomial_eval.js
const start = Date.now();
const coeffs = [];
for (let i = 0; i <= 1000; i++) coeffs.push(i + 1);
function horner(coeffs, x) {
    let result = coeffs[coeffs.length - 1];
    for (let i = coeffs.length - 2; i >= 0; i--) {
        result = result * x + coeffs[i];
    }
    return result;
}
let s = 0;
for (let i = 0; i < 100000; i++) {
    s += horner(coeffs, 1.5);
}
const end = Date.now();
console.log(`Sum: ${s}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/src/ruby/polynomial_eval.rb
coeffs = (1..1001).to_a
start = Time.now.to_f * 1000
def horner(coeffs, x)
    result = coeffs[-1]
    (coeffs.length - 2).downto(0) do |i|
        result = result * x + coeffs[i]
    end
    result
end
s = 0
100000.times { s += horner(coeffs, 1.5) }
finish = Time.now.to_f * 1000
puts "Result: #{s}"
puts "Time: #{(finish - start).round}ms"

